Sources/WalletConnectModal/Modal/ModalSheet.swift

```diff
@@ -81,14 +88,31 @@ public struct ModalSheet: View {
         .overlay(
             VStack {
                 if viewModel.destination.hasSearch {
-                    TextField("Search", text: $viewModel.searchTerm)
-                        .transform {
-                            #if os(iOS)
-                            $0.textFieldStyle(.roundedBorder)
-                                .autocapitalization(.none)
-                            #endif
-                        }
-                        .padding(.horizontal, 50)
+                    
+                    HStack {
+                        Image(systemName: "magnifyingglass")
+                        TextField("Search", text: $viewModel.searchTerm, onEditingChanged: { editing in
+                            self.searchEditing = editing
+                        })
+                    }
+                    .padding(.vertical, 4)
+                    .padding(.horizontal, 10)
+                    .background(Color.background3)
+                    .foregroundColor(searchEditing ? .foreground1 : .foreground3)
+                    .cornerRadius(28)
+                    .overlay(
+                        RoundedRectangle(cornerRadius: 28)
+                            .stroke(searchEditing ? Color.accent : Color.thinOverlay, lineWidth: 1)
+                    )
+                    .onDisappear {
+                        searchEditing = false
+                    }
+                    .transform {
+                        #if os(iOS)
+                            $0.autocapitalization(.none)
```

Comment:
why autocapitalization is inside transform?

Reply:
For some reason, `autocapitalization` is not available on macOS, so I had to wrap it in this `#if os()` and this `transform` helper method allows to do that inline rather than having the entire view duplicated with `if/else`.

However I've change the condition to really reflect the problem, and add `#if not macOS`

Reply:
The transform method is really this simple
```swift
func transform(@ViewBuilder _ transform: (Self) -> some View) -> some View {
        transform(self)
    }
```